### PR TITLE
BaSP-MR-103: Add requested fixes

### PR DESCRIPTION
**Pull request by SantiH97**
-Got the modals out of the mapping section of the render
-Changed slots from input to select
-Added error handling
-Fixed the PUT method
-Fixed the POST method
-Made the modal messeges more accurate

### CODE REVIEW

File: src/Components/Classes/Form/index.jsx [hunk 1]
```diff
@@ -185,6 +299,7 @@ const Form = () => {
   };
 
   const updateClass = async (id, newClass) => {
+    console.log(newClass);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:302]: Remove console.log

File: src/Components/Classes/Form/index.jsx [hunk 2]
```diff
@@ -193,42 +308,84 @@ const Form = () => {
         },
         body: JSON.stringify(newClass)
       });
-      await res.json();
+      const data = await res.json();
+      console.log(res.status);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:312]: Remove console.log

File: src/Components/Classes/Form/index.jsx [hunk 3]
```diff
@@ -171,7 +276,16 @@ const Form = () => {
           'Content-type': 'application/json'
         }
       });
-      await res.json();
+      const data = await res.json();
+      console.log(newClass);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:280]: Remove console.log

File: src/Components/Classes/Form/index.jsx [hunk 4]
```diff
@@ -193,42 +308,84 @@ const Form = () => {
...
         body: JSON.stringify(newClass)
       });
-      await res.json();
+      const data = await res.json();
+      console.log(res.status);
+      if (res.status === 200) {
+        setSuccess(true);
+        setShowSuccessModal(true);
+      } else {
+        console.log(data);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:317]: Remove console.log

File: src/Components/Classes/Form/index.jsx [hunk 5]
```diff
@@ -193,42 +308,84 @@ const Form = () => {
...
+        setSuccess(false);
+        setShowErrorModal(true);
+      }
     } catch (error) {
       console.error(error);
     }
   };
 
+  const previousClass = (singleClass) => {
+    console.log(singleClass);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:328]: Remove console.log

File: src/Components/Classes/Form/index.jsx [hunk 6]
```diff
@@ -193,42 +308,84 @@ const Form = () => {
...
+      setNewClass({
+        ...newClass,
+        day: singleClass.day,
+        hour: singleClass.hour,
+        activityId: singleClass.activityId._id,
+        trainerId: singleClass.trainerId._id,
+        slots: singleClass.slots
+      });
+    console.log(newClass);
+    console.log(message);
```
Comment (DelfiLT) [inline src/Components/Classes/Form/index.jsx:339]: Remove both console.log

**Review by DelfiLT — APPROVED** [review]

**Review by Bigzufi — APPROVED** [review]
Good Job!

**Review by Milicerro — COMMENTED** [review]
The Update function does not work, when you click the update icon the app crashes.

**Pull request merged**